ambry-router/src/main/java/com.github.ambry.router/DeleteManager.java

```diff
@@ -100,10 +100,11 @@ void submitDeleteBlobOperation(String blobIdString, FutureResult<Void> futureRes
     try {
       BlobId blobId = RouterUtils.getBlobIdFromString(blobIdString, clusterMap);
       DeleteOperation deleteOperation =
-          new DeleteOperation(routerConfig, responseHandler, blobId, futureResult, callback, time);
+          new DeleteOperation(routerConfig, responseHandler, blobId, futureResult, callback, time, routerMetrics);
       deleteOperations.add(deleteOperation);
     } catch (RouterException e) {
       routerMetrics.operationDequeuingRate.mark();
+      routerMetrics.countError(e, NonBlockingRouter.RouterOperationType.DeleteBlob);
```

Comment:
I don't want you to change your preference :) I am trying to convince you that this is extra code to achieve the same thing in a roundabout way.
The chances of someone setting `putErrorRate` instead of `deleteErrorRate` is the same as passing in `Put` type instead of `Delete` type to that method.